### PR TITLE
feat: AdminAPI accepts VerifiedAddresses

### PR DESCRIPTION
**Pull request by austin-king**
## High Level Overview of Change

AdminAPI now accepts a POST containing `VerifiedAddresses`.

### Context of Change

https://app.asana.com/0/1182225689526777/1186929984905885

There is one final PR regarding admin api (PUT functionality) for v. payID. 

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Before / After

AdminAPI now accepts verified addresses when posting a user.

## Test Plan

Previous tests continue to work + I added one confirming it gives us a `201` upon posting a user w/ verified addresses.

## Future Tasks

PUT functionality.
-->

### CODE REVIEW

File: src/middlewares/users.ts [hunk 1]
```diff
@@ -113,7 +114,16 @@ export async function postUser(
   // TODO:(hbergren) Need to test here and in `putUser()` that `req.body.addresses` is well formed.
   // This includes making sure that everything that is not ACH or ILP is in a CryptoAddressDetails format.
   // And that we `toUpperCase()` paymentNetwork and environment as part of parsing the addresses.
-  await insertUser(payId, req.body.addresses, req.body.identityKey)
+  let allAddresses: AddressInformation[] = []
+
+  if (req.body.addresses !== undefined) {
```
Comment (austin-king) [inline src/middlewares/users.ts:119]: When we put type validation into this middleware these checks can likely be improved

Comment (hbergren) [reply src/middlewares/users.ts:119]: What you have is probably fine.

You could also do something like this:
```ts
const allAddresses = [...req.body.addresses || [], ...req.body.verifiedAddresses || []]
```

Typing that out, I think what you have is more readable though haha.

File: test/integration/e2e/admin-api/postUsers.test.ts [hunk 2]
```diff
@@ -48,6 +48,44 @@ describe('E2E - adminApiRouter - POST /users', function (): void {
...
+    // WHEN we make a POST request to /users with that user information
+    request(app.adminApiExpress)
+      .post(`/users`)
+      .set('PayID-API-Version', payIdApiVersion)
+      .send(userInformation)
+      .expect('Content-Type', /text\/plain/u)
+      // THEN we expect the Location header to be set to the path of the created user resource
+      .expect('Location', `/users/${userInformation.payId}`)
+      // THEN we expect back a 201 - CREATED
+      .expect(HttpStatus.Created, done)
```
Comment (hbergren) [inline test/integration/e2e/admin-api/postUsers.test.ts:86]: Is it worth having a subsequent GET request to verify that it the PayID actually saved the `verifiedAddresses`?

File: src/middlewares/users.ts (hunk 1 above)
Comment (austin-king) [reply src/middlewares/users.ts:119]: Ya I like that but I'll just keep it as is

File: test/integration/e2e/admin-api/postUsers.test.ts (hunk 2 above)
Comment (austin-king) [reply test/integration/e2e/admin-api/postUsers.test.ts]: Great point, just pushed a commit stringing a GET to confirm afterwards.

**Pull request merged**